hazelcast-jet-core/src/main/java/com/hazelcast/jet/Processor.java

```diff
@@ -144,5 +145,29 @@ default boolean isCooperative() {
          */
         @Nonnull
         String vertexName();
+
+        /**
+         * Returns the future to check for cancellation status.
+         * <p>
+         * This is only necessary, if the {@link #complete()} method never returns and blocks
```

Comment:
This is not the only case, it is enough that the processor does more than one blocking call per processor call.